association_backend/association/api/views.py

```diff
@@ -1,18 +1,45 @@
 import logging
-import typing
+from typing import Optional, Union
 
-from association.api.context import Context
-from association.api.schema import schema
 from starlette.requests import Request
+from starlette.responses import Response
+from starlette.types import Receive, Scope, Send
 from starlette.websockets import WebSocket
 from strawberry.asgi import GraphQL as BaseGraphQL
 
+from association.api.context import Context
+from association.api.schema import schema
+from association.settings import PASTAPORTO_ACTION_SECRET, PASTAPORTO_ACTION_X_HEADER
+
 logger = logging.getLogger(__name__)
 
 
 class GraphQL(BaseGraphQL):
     def __init__(self) -> None:
         super().__init__(schema)
 
-    async def get_context(self, request: typing.Union[Request, WebSocket]) -> Context:
+    async def handle_http(self, scope: Scope, receive: Receive, send: Send):
+        request = Request(scope=scope, receive=receive)
+        root_value = await self.get_root_value(request)
+        context = await self.get_context(request)
+
+        response = await self.get_http_response(
+            request=request,
+            execute=self.execute,
+            process_result=self.process_result,
+            graphiql=self.graphiql,
+            root_value=root_value,
+            context=context,
+        )
+
+        if context.pastaporto_action:
+            response.headers[
+                PASTAPORTO_ACTION_X_HEADER
+            ] = context.pastaporto_action.sign(PASTAPORTO_ACTION_SECRET)
+
+        await response(scope, receive, send)
```

Comment:
? you don't need all of this? this service doesn't send (yet?) actions back to the gateway

Reply:
@marcoacierno 
If I understand correctly I have to delete `GraphQL.handle_http` method and `Context.pastaporto_action` variable, right?

Reply:
ok done